caas/kubernetes/provider/application/application.go

```diff
@@ -110,7 +112,7 @@ func newApplication(
 	clock clock.Clock,
 	randomPrefix k8sutils.RandomPrefixFunc,
 	newApplier func() resources.Applier,
-) caas.Application {
+) *app {
```

Comment:
Do we need this?
Is it just for making tests work? Could we cast to *app in the test helper?

Reply:
this `newApplication` was created for test initially because it will be wrapped up in `NewApplication`(it returns `caas.Application`).